lib/luacrypto_skcipher.c

```diff
@@ -0,0 +1,277 @@
+		lunatik_free(tfm_ud->work_buffer);
+		tfm_ud->work_buffer = lunatik_checkalloc(L, data_s_len);
+		tfm_ud->work_buffer_len = data_s_len;
+	}
+	memcpy(tfm_ud->work_buffer, data_s, data_s_len);
+
+	sg_init_one(&tfm_ud->sg_work, tfm_ud->work_buffer, data_s_len);
+
+	skcipher_request_set_crypt(tfm_ud->req, &tfm_ud->sg_work, &tfm_ud->sg_work,
+				   data_s_len, tfm_ud->iv_data);
+	skcipher_request_set_callback(tfm_ud->req, 0, NULL, NULL);
+
+	if (encrypt) {
+		lunatik_try(L, crypto_skcipher_encrypt, tfm_ud->req);
+	} else {
+		lunatik_try(L, crypto_skcipher_decrypt, tfm_ud->req);
+	}
+
+	lua_pushlstring(L, (const char *)tfm_ud->work_buffer, data_s_len);
+	return 1;
+}
+
+/***
+* Encrypts plaintext using the SKCIPHER transform.
+* The IV (nonce) must be unique for each encryption operation with the same key for most modes.
+* Plaintext length should be appropriate for the cipher mode (e.g., multiple of blocksize).
+* @function crypto_skcipher:encrypt
+* @tparam string iv The Initialization Vector. Its length must match `ivsize()`.
+* @tparam string plaintext The data to encrypt.
+* @treturn string The ciphertext.
+* @raise Error on encryption failure, incorrect IV length, or allocation issues.
+*/
+static int luacrypto_skcipher_encrypt(lua_State *L) {
+	return luacrypto_skcipher_crypt_common(L, true);
+}
+
+/***
+* Decrypts ciphertext using the SKCIPHER transform.
+* The IV must match the one used during encryption.
+* Ciphertext length should be appropriate for the cipher mode.
+* @function crypto_skcipher:decrypt
+* @tparam string iv The Initialization Vector. Its length must match `ivsize()`.
+* @tparam string ciphertext The data to decrypt.
+* @treturn string The plaintext.
+* @raise Error on decryption failure, incorrect IV length, or allocation issues.
+*/
+static int luacrypto_skcipher_decrypt(lua_State *L) {
+	return luacrypto_skcipher_crypt_common(L, false);
+}
+/***
+* Lua C methods for the SKCIPHER TFM object.
+* Includes cryptographic operations and Lunatik metamethods.
+* The `__close` method is important for explicit resource cleanup.
+* @see crypto_skcipher
+* @see lunatik_closeobject
+*/
+
+static const luaL_Reg luacrypto_skcipher_mt[] = {
+	{"setkey", luacrypto_skcipher_setkey},
+	{"ivsize", luacrypto_skcipher_ivsize},
+	{"blocksize", luacrypto_skcipher_blocksize},
+	{"encrypt", luacrypto_skcipher_encrypt},
+	{"decrypt", luacrypto_skcipher_decrypt},
+	{"__gc", lunatik_deleteobject},
+	{"__close", lunatik_closeobject},
+	{"__index", lunatik_monitorobject},
+	{NULL, NULL}
+};
+
+/***
+* Lunatik class definition for SKCIPHER TFM objects.
+* This structure binds the C implementation (luacrypto_skcipher_priv, methods, release function)
+* to the Lua object system managed by Lunatik.
+*/
+
+static const lunatik_class_t luacrypto_skcipher_class = {
+	.name = "crypto_skcipher",
+	.methods = luacrypto_skcipher_mt,
+	.release = luacrypto_skcipher_release,
+	.sleep = true,
+};
+
+
```

Comment:
```suggestion

```

let's please avoid multiple `\n`.. I just noticed we let it pass through review on luacrypto_shash